integrationtest/integration_test.go

```diff
@@ -220,6 +243,84 @@ func TestFailWithBadTokenReviewerJwt(t *testing.T) {
 	}
 }
 
+func TestAuthAliasMetadataAssignment(t *testing.T) {
+	// annotate the service account
+	expMetadata := map[string]string{
+		"foo": "bar",
+		"bar": "baz",
+	}
+
+	annotationPrefix := "auth-metadata.vault.hashicorp.com/"
+	annotations := map[string]string{}
+	for k, v := range expMetadata {
+		annotations[annotationPrefix+k] = v
+	}
+	annotateServiceAccount(t, "vault", annotations)
+
+	client, cleanup := setupKubernetesAuth(t, "vault", nil, nil)
+	defer cleanup()
+
+	loginSecret, err := client.Logical().Write("auth/kubernetes/login", map[string]interface{}{
```

Comment:
This test looks good. One further improvement we could make though (and I think it would probably also be a net simplification) is to replace the entity API enumeration code with a straightforward secret-retrieval call that relies on a templated policy attached to the token we get from this call.

Reply:
Thank you. I'll change the flow to use templated policies.